aiida_cp2k/calculations.py

```diff
@@ -0,0 +1,214 @@
+    _DEFAULT_PARENT_CALC_FLDR_NAME = "parent_calc/"
+    _DEFAULT_COORDS_FILE_NAME = "aiida.coords.xyz"
+    _DEFAULT_PARSER = "cp2k"
+
+    @classmethod
+    def define(cls, spec):
+        super(Cp2kCalculation, cls).define(spec)
+
+        # Input parameters
+        spec.input("parameters", valid_type=Dict, help="the input parameters")
+        spec.input(
+            "structure",
+            valid_type=StructureData,
+            required=False,
+            help="the input structure",
+        )
+        spec.input(
+            "settings",
+            valid_type=Dict,
+            required=False,
+            help="additional input parameters",
+        )
+        spec.input(
+            "resources", valid_type=dict, required=False, help="special settings"
+        )
+        spec.input(
+            "parent_calc_folder",
+            valid_type=RemoteData,
+            required=False,
+            help="remote folder used for restarts",
+        )
+        spec.input_namespace(
+            "file",
+            valid_type=SinglefileData,
+            required=False,
+            help="additional input files",
+            dynamic=True,
+        )
+
+        # Default file names, parser, etc..
+        spec.input(
+            "metadata.options.input_filename",
+            valid_type=six.string_types,
+            default=cls._DEFAULT_INPUT_FILE,
+            non_db=True,
+        )
+        spec.input(
+            "metadata.options.output_filename",
+            valid_type=six.string_types,
+            default=cls._DEFAULT_OUTPUT_FILE,
+            non_db=True,
+        )
+        spec.input(
+            "metadata.options.parser_name",
+            valid_type=six.string_types,
+            default=cls._DEFAULT_PARSER,
+            non_db=True,
+        )
+
+        # Exit codes
+        spec.exit_code(
+            100,
+            "ERROR_NO_RETRIEVED_FOLDER",
+            message="The retrieved folder data node could not be accessed.",
+        )
+
+        # Output parameters
+        spec.output(
+            "output_parameters",
+            valid_type=Dict,
+            required=True,
+            help="the results of the calculation",
+        )
+        spec.output(
+            "output_structure",
+            valid_type=StructureData,
+            required=False,
+            help="optional relaxed structure",
+        )
+        spec.output(
+            "output_bands",
+            valid_type=BandsData,
+            required=False,
+            help="optional band structure",
+        )
+
+    # --------------------------------------------------------------------------
+    def prepare_for_submission(self, folder):
+        """Create the input files from the input nodes passed to this instance of the `CalcJob`.
+
+        :param folder: an `aiida.common.folders.Folder` to temporarily write files on disk
+        :return: `aiida.common.datastructures.CalcInfo` instance
+        """
+        from .utils import Cp2kInput
```

Comment:
```suggestion
        from aiida_cp2k.utils import Cp2kInput
```

Reply:
Apart from that change I think the commit 0acf6db0deee6444c812ffdb313c9d634fdac04b is good to go.

I will continue checking next commits (probably some of them can be squashed in)